### DISCUSSION TITLE
Downloading a DASH video in 4k, video downloads with the worst quality

### DISCUSSION THREAD

**Discussion by jamytamy32** [aclap-dev/video-downloadhelper#477] [Help and Q&A]
### Your question

Hello everyone, I'm trying to download a video from the adult site teamskeet, I had no issues in the past but now when I try to download a video in 4k it downloads with the worst quality, the video itself is the full length but the issue is the quality.

### Addon & CoApp version and details

version: 8.2.2.9
channel: stable
build date: 2024-03-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: es-MX
license: accepted
key (only 16 first characters): 221e2a390df946c3
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\ADMIN",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by paulrouget** [comment]
You're correct. I'll look into it.

**Reply by jamytamy32** [reply]
Thanks!

**Comment by GrampaWildWilly** [comment]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3447, Firefox 125.0.3 64-bit, licensed VDH 9.0.1.10a1 beta, CoApp 2.0.19.  I do not work for VDH.  I'm just another user.

I also am not a subscriber to TeamSkeet.  However, their site allows you to go to pages where full videos would be hosted if you logged in.  On those pages, instead of the full videos, they offer trailers that you can play & download.  I picked a page at random & looked at what VDH was offering.  Among the various items on the VDH menu was one labelled MPD & duration 2:22.  There were several other entries on the VDH menu but you mentioned DASH as being the problem so I focused on this MPD entry.  VDH advertised a 1080p variant for this trailer so I downloaded it.  Sure enough, I got something that was 426x240.

I thought let's try something outside the box.  I reloaded VDH & reloaded the web page.  But this time, I decided to change the resolution on the trailer from 1080p to 720p.  What I downloaded is what you can see in this ffprobe report.

[ffprobe TeamSkeet.txt](https://github.com/aclap-dev/video-downloadhelper/files/15287897/ffprobe.TeamSkeet.txt)

Weirdest thing I've ever seen.  I did not get a 720p video.  I got this thing that consists of TWO video tracks & NO audio tracks.  When I played it in VLC, it played the 1080p video track on one of my monitors & the 240p video track on my other monitor.  That is just bizarre.

Now, it looks like for this DASH content, no matter what resolution you pick, VDH downloads a different resolution.  If you leave VDH to download the default resolution, which looks like it's going to be 1080p, it instead downloads the 240p variant . . . with a valid audio track.  If you don't let VDH choose a default resolution, VDH still doesn't download the resolution you select, but it looks like it chooses another video resolution for the audio track.  That is way weird.

Just to prove that it is possible to download this trailer properly . . .

[TeamSkeet Trailer mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/15287911/TeamSkeet.Trailer.mp4.Log.txt)

This did retrieve the 1920x1080 video with its proper sound track.  That is despite the weird 404 Not Found error that apparently terminated the download.  I don't know what's up with that.  Despite that error, the video did download & the full 2:22 of it did play in VLC with both video & audio.  I don't know what it was trying to get at the end.  Maybe the DASH master manifest tries to continue with something else that just isn't available unless you are a member & logged in.  Or maybe the manifest actually has an error in it of some kind.  Whatever the case, the error seems to have occurred after the full download has completed successfully, giving a perfectly usable result file.

I don't know if this helps you any @paulrouget but this is what I discovered.

**Reply by GrampaWildWilly** [reply]
Oh.  MPD.  Paul, that's not consistent.  For m3u8 manifests, the VDH menu entries mention HLS, which is correct.  No VDH menu entry says m3u8.  To make things consistent, the VDH menu entry for this TeamSkeet trailer I found should say DASH, not MPD.

**Reply by paulrouget** [reply]
Yes it's MPD quality discovery bug. On my radar.

**Reply by dastels** [reply]
This started being a problem when Teamskeet updated their site/ui.
